tests/returntypes/test_torch_qnode_new.py

```diff
@@ -0,0 +1,2138 @@
+
+        assert np.array_equal(result[0].shape, (n_sample,))
+        assert result[1].shape == ()
+        assert isinstance(result[1], torch.Tensor)
+        assert result[2].shape == ()
+        assert isinstance(result[2], torch.Tensor)
+        assert result[0].dtype is torch.int64
+
+    def test_single_wire_sample(self, tol):
+        """Test the return type and shape of sampling a single wire"""
+        n_sample = 10
+
+        dev = qml.device("default.qubit", wires=1, shots=n_sample)
+
+        @qnode(dev, diff_method="parameter-shift", interface="torch")
+        def circuit():
+            qml.RX(0.54, wires=0)
+            return qml.sample(qml.PauliZ(0))
+
+        result = circuit()
+
+        assert isinstance(result, torch.Tensor)
+        assert np.array_equal(result.shape, (n_sample,))
+
+    def test_multi_wire_sample_regular_shape(self, tol):
+        """Test the return type and shape of sampling multiple wires
+        where a rectangular array is expected"""
+        n_sample = 10
+
+        dev = qml.device("default.qubit", wires=3, shots=n_sample)
+
+        @qnode(dev, diff_method="parameter-shift", interface="torch")
+        def circuit():
+            return qml.sample(qml.PauliZ(0)), qml.sample(qml.PauliZ(1)), qml.sample(qml.PauliZ(2))
+
+        result = circuit()
+
+        # If all the dimensions are equal the result will end up to be a proper rectangular array
+        assert isinstance(result, tuple)
+        assert tuple(result[0].shape) == (n_sample,)
+        assert tuple(result[1].shape) == (n_sample,)
+        assert tuple(result[2].shape) == (n_sample,)
+        assert result[0].dtype == torch.int64
+        assert result[1].dtype == torch.int64
+        assert result[2].dtype == torch.int64
+
+
+qubit_device_and_diff_method_and_mode = [
+    ["default.qubit", "backprop", "forward"],
+    ["default.qubit", "finite-diff", "backward"],
+    ["default.qubit", "parameter-shift", "backward"],
+    ["default.qubit", "adjoint", "forward"],
+    ["default.qubit", "adjoint", "backward"],
+]
+
+
+@pytest.mark.parametrize("dev_name,diff_method,mode", qubit_device_and_diff_method_and_mode)
+@pytest.mark.parametrize("shots", [None, 10000])
+class TestReturn:
+    """Class to test the shape of the Grad/Jacobian/Hessian with different return types."""
+
+    def test_grad_single_measurement_param(self, dev_name, diff_method, mode, shots):
+        """For one measurement and one param, the gradient is a float."""
+        if shots is not None and diff_method in ("backprop", "adjoint"):
+            pytest.skip("Test does not support finite shots and adjoint/backprop")
+
+        dev = qml.device(dev_name, wires=1, shots=shots)
+
+        @qnode(dev, interface="torch", diff_method=diff_method, mode=mode)
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        a = torch.tensor(0.1, requires_grad=True)
+
+        res = circuit(a)
```

Comment:
Minor: should `res` type and shape be checked too? E.g., to make sure the output is squeezed.

Reply:
Added.